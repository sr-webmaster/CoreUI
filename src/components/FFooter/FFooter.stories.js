import { MAIN } from '../../../.storybook/categories'
import FFooter from './FFooter.vue'

export default {
  title: `${MAIN}|Footer`,
  id: 'FFooter',
  description: 'Hello'
}

/**
 * https://storybook.js.org/docs/guides/guide-vue/#step-4-write-your-stories
 */
export const defaultStory = () => ({
  components: {
    FFooter
  },
  template: `
    <f-footer></f-footer>
  `
})

export const PrimaryColor = () => ({
  components: {
    FFooter
  },
  template: `
    <f-footer color="primary"></f-footer>
  `
})

export const SecondaryColor = () => ({
  components: {
    FFooter
  },
  template: `
    <f-footer color="secondary"></f-footer>
  `
})

export const AccentColor = () => ({
  components: {
    FFooter
  },
  template: `
    <f-footer color="accent"></f-footer>
  `
})
