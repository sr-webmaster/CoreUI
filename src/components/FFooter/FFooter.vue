<template>
  <v-footer
    :color="color"
    class="px-3 white--text"
    app
  >
    <v-layout
      row
      wrap
    >
      <span>Powered by FreshinUp</span>
      <v-spacer />
      <span>&copy; 2020</span>
    </v-layout>
  </v-footer>
</template>

<script>
export default {
  props: {
    color: {
      type: String,
      default: 'accent',
      validator (value) {
        return ['primary', 'secondary', 'accent'].indexOf(value) !== -1
      }
    }
  }
}
</script>

<style scoped>

</style>
