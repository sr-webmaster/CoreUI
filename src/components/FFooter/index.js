import Component from './FFooter'
export default Component
