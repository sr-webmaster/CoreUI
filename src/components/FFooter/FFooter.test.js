import { mount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import { defaultStory, PrimaryColor, SecondaryColor } from './FFooter.stories'

describe('FFooter', () => {
  describe('Computed', () => {
    // None
  })

  describe('Methods', () => {
    // None
  })

  describe('Visuals', () => {
    test('defaults', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(defaultStory(), {
        localVue: createLocalVue().vue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.find('.v-footer').classes()).toContain('accent')
      expect(wrapper.element).toMatchSnapshot()
    })

    test('footColor property (primary)', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(PrimaryColor(), {
        localVue: createLocalVue().vue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.find('.v-footer').classes()).toContain('primary')
      expect(wrapper.element).toMatchSnapshot()
    })

    test('footColor property (secondary)', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(SecondaryColor(), {
        localVue: createLocalVue().vue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.find('.v-footer').classes()).toContain('secondary')
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
