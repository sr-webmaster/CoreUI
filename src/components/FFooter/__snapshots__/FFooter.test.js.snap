// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FFooter Visuals defaults 1`] = `
<footer
  class="v-footer px-3 white--text v-footer--fixed theme--light accent"
  style="height: 32px;"
>
  <div
    class="layout row wrap"
  >
    <span>
      Powered by FreshinUp
    </span>
     
    <div
      class="spacer"
    />
     
    <span>
      © 2020
    </span>
  </div>
</footer>
`;

exports[`FFooter Visuals footColor property (primary) 1`] = `
<footer
  class="v-footer px-3 white--text v-footer--fixed theme--light primary"
  style="height: 32px;"
>
  <div
    class="layout row wrap"
  >
    <span>
      Powered by FreshinUp
    </span>
     
    <div
      class="spacer"
    />
     
    <span>
      © 2020
    </span>
  </div>
</footer>
`;

exports[`FFooter Visuals footColor property (secondary) 1`] = `
<footer
  class="v-footer px-3 white--text v-footer--fixed theme--light secondary"
  style="height: 32px;"
>
  <div
    class="layout row wrap"
  >
    <span>
      Powered by FreshinUp
    </span>
     
    <div
      class="spacer"
    />
     
    <span>
      © 2020
    </span>
  </div>
</footer>
`;
